src/celeritas/neutron/interactor/NeutronElasticInteractor.hh

```diff
@@ -0,0 +1,155 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/interactor/NeutronElasticInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/math/ArrayOperators.hh"
+#include "celeritas/Constants.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/mat/IsotopeView.hh"
+#include "celeritas/neutron/data/NeutronElasticData.hh"
+#include "celeritas/phys/FourVector.hh"
+#include "celeritas/phys/Interaction.hh"
+#include "celeritas/phys/ParticleTrackView.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+#include "detail/InvariantQ2Sampler.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Perform neutron elastic scattering based on the CHIPS (Chiral invariant
+ * phase space) model.
+ *
+ * \note This performs the sampling procedure as in G4HadronElastic,
+ * G4ChipsElasticModel and G4ChipsNeutronElasticXS, as partly documented
+ * in section 21.1.3 of the Geant4 Physics Reference (release 11.2).
```

Comment:
Can you add a description of the implementation here, especially if the documentation in the reference model is "partial"? Include:
- Discussion of translation between lab-frame and COM-frame using FourVector and boost
- Link to COM-frame scatter direction calculator (Q2)
- Resulting neutron and deposition energy are kinematically calculated from the scattering direction